fe/src/main/java/org/apache/doris/analysis/DateLiteral.java

```diff
@@ -319,7 +319,7 @@ protected void toThrift(TExprNode msg) {
 
     @Override
     protected Expr uncheckedCastTo(Type targetType) throws AnalysisException {
-        if (targetType.isDateType()) {
+        if (targetType.equals(this.type)) {
```

Comment:
Why not return a new DateLiteral when targetType and this.type are different but both dateType.